# Bonus: Temperature Analysis I

In [4]:
import pandas as pd
from datetime import datetime as dt
from scipy import stats

In [5]:
# "tobs" is "temperature observations"

hawaii_df = pd.read_csv('Resources/hawaii_measurements.csv')
hawaii_df.head()

station        date  prcp  tobs
0  USC00519397  2010-01-01  0.08    65
1  USC00519397  2010-01-02  0.00    63
2  USC00519397  2010-01-03  0.00    74
3  USC00519397  2010-01-04  0.00    76
4  USC00519397  2010-01-06   NaN    73

In [6]:
# Convert the date column format from string to datetime
hawaii_df.date = pd.to_datetime(hawaii_df.date, infer_datetime_format=True)

In [7]:
# Set the date column as the DataFrame index
hawaii_df = hawaii_df.set_index(hawaii_df['date'])
hawaii_df.head()

station       date  prcp  tobs
date                                          
2010-01-01  USC00519397 2010-01-01  0.08    65
2010-01-02  USC00519397 2010-01-02  0.00    63
2010-01-03  USC00519397 2010-01-03  0.00    74
2010-01-04  USC00519397 2010-01-04  0.00    76
2010-01-06  USC00519397 2010-01-06   NaN    73

In [8]:
# Drop the date column
hawaii_df = hawaii_df.drop(columns='date')
hawaii_df.head()

station  prcp  tobs
date                               
2010-01-01  USC00519397  0.08    65
2010-01-02  USC00519397  0.00    63
2010-01-03  USC00519397  0.00    74
2010-01-04  USC00519397  0.00    76
2010-01-06  USC00519397   NaN    73

### Compare June and December data across all years 

In [9]:
from scipy import stats

In [10]:
# Filter data for desired months
juneData = hawaii_df[hawaii_df.index.month == 6]
juneData.head()

station  prcp  tobs
date                               
2010-06-01  USC00519397  0.00    78
2010-06-02  USC00519397  0.01    76
2010-06-03  USC00519397  0.00    78
2010-06-04  USC00519397  0.00    76
2010-06-05  USC00519397  0.00    77

In [11]:
decemberData = hawaii_df[hawaii_df.index.month == 12]
decemberData.head()

station  prcp  tobs
date                               
2010-12-01  USC00519397  0.04    76
2010-12-03  USC00519397  0.00    74
2010-12-04  USC00519397  0.00    74
2010-12-06  USC00519397  0.00    64
2010-12-07  USC00519397  0.00    64

In [12]:
# Identify the average temperature for June
juneData.mean()

prcp     0.136360
tobs    74.944118
dtype: float64

In [13]:
# Identify the average temperature for December
decemberData.mean()

prcp     0.216819
tobs    71.041529
dtype: float64

In [14]:
# Create collections of temperature data
juneTemp = juneData.tobs
juneTemp

date
2010-06-01    78
2010-06-02    76
2010-06-03    78
2010-06-04    76
2010-06-05    77
              ..
2017-06-26    79
2017-06-27    74
2017-06-28    74
2017-06-29    76
2017-06-30    75
Name: tobs, Length: 1700, dtype: int64

In [15]:
decemberTemp = decemberData.tobs
decemberTemp

date
2010-12-01    76
2010-12-03    74
2010-12-04    74
2010-12-06    64
2010-12-07    64
              ..
2016-12-27    71
2016-12-28    71
2016-12-29    69
2016-12-30    65
2016-12-31    65
Name: tobs, Length: 1517, dtype: int64

In [16]:
stats.ttest_ind(juneTemp,decemberTemp)

Ttest_indResult(statistic=31.60372399000329, pvalue=3.9025129038616655e-191)

### Analysis

The mean temperature difference between the June and December is 3.9 degrees Fahrenheit which isn't much of a difference.
The t-test has a extremely low p-value indicating that the difference is statistically significant. 
While the difference is meaningful, it isn't  indicating that you can travel to Hawaii and enjoy 70 degree temperature year-round!¶